### DISCUSSION TITLE
The marketplace for Front Matter

### DISCUSSION THREAD

**Discussion by estruyf** [estruyf/vscode-front-matter#420] [General]
More and more, the idea comes to my mind to build a marketplace that integrates with the website and extension. 

Having a tool/product is great, but having the ability to extend it to your own needs is even better. I believe Front Matter already offers many of these extensibility options like custom actions/scripts, dynamic placeholders, and more.

With these extensibilities, I started thinking about a way to make it easier to share your extensibility with others. I believe a marketplace is a great opportunity for this.

### The package

An extensibility package would contain the following files:

- readme
- script file
- `extensibility.json`: The file with details about the current extensibility
  - Name
  - Type
  - Version
  - Author
  - postInstall: A script to start an installation or script after the extensibility installation to your project

### Installation

Installation will happen to the `.frontmatter/extensibility` folder. 

### Your thoughts and ideas

What do you think?

**Comment by michaeltlombardi** [comment]
This is nearly exactly the solution to the problem domain I'm working in for providing a delightful experience for users of my Hugo modules. I currently publish (beta) configs users can reference in their `frontMatter.extends` setting, but that doesn't handle scripts and relies on URL for versioning.

This would be a much more discoverable and maintainable way for me to publish the Front Matter integration for my users.
